tracing-attributes/src/expand.rs

```diff
@@ -440,8 +440,12 @@ fn param_names(pat: Pat, record_type: RecordType) -> Box<dyn Iterator<Item = (Id
 enum AsyncTraitKind<'a> {
     // old construction. Contains the function
     Function(&'a ItemFn),
-    // new construction. Contains a reference to the async block
-    Async(&'a ExprAsync),
+    // new construction. Contains a reference to the async block, and a bool
+    // indicating if the return value is a `Box::pin`
+    Async {
+        async_expr: &'a ExprAsync,
+        pinned_box: bool,
+    },
 }
 
 pub(crate) struct AsyncTraitInfo<'block> {
```

Comment:
similarly, it seems like this is now used for all functions returning `async` blocks, not just those generated by `async-trait`. we should probably rename it.